packages/ecdh/tests/ecdh.tests.ts

```diff
@@ -0,0 +1,68 @@
+import { BarretenbergBackend } from '@noir-lang/backend_barretenberg'
+import { Noir } from '@noir-lang/noir_js'
+import { compile } from '@noir-lang/noir_wasm'
+import { ProofData } from '@noir-lang/types'
+import { expect } from 'chai'
+import { randomBytes } from 'crypto'
+import { readFileSync } from 'fs'
+import { resolve } from 'path'
+import 'mocha'
+import circuit_ecdh from '../../ecdh/target/ecdh.json'
+
+
+
+function generatePrivateKey(): Uint8Array {
+  return randomBytes(32)
+}
+
+describe('ECDH Circuit Tests', function() {
+  let noir: Noir;
+  let backend: BarretenbergBackend;
+  let correctProof: ProofData;
+  beforeEach(async () => {
+    const circuitFile = readFileSync(resolve(__dirname, '../target/ecdh.json'), 'utf-8')
+    const circuit = JSON.parse(circuitFile)
+    backend = new BarretenbergBackend(circuit)
+    noir = new Noir(circuit, backend)
+  })
+  it('Should generate valid proof for correct input', async function() {
+    this.timeout(20000); // Optional: Increase timeout if needed
+    let pk1 = generatePrivateKey();
+    let pk2 = generatePrivateKey();
+
+    // Convert Uint8Array to regular arrays
+    const input = {
+      private_key1: Array.from(pk1),
+      private_key2: Array.from(pk2),
+    }
+
+    try {
+      correctProof = await noir.generateProof(input)
+      console.log('Proof generated:', correctProof)
+    } catch (error) {
+      console.error('Error generating proof:', error)
+      throw error // Ensure errors are not swallowed
+    }
```

Comment:
Avoid using try/catch in a test. You want to test your code as you wrote it: try/catch belongs rather to src code to handle error at runtime.

In your tests you can totally (sometimes even recommended) cover both cases: no error and error. 
But the best practice is to keep the tests separated:
- 1 test for the normal, success case
- 1 test for the error (that you trigger with some "setup" statements specific to that single test) 
  e.g you can use [`should.throw`](https://www.chaijs.com/api/bdd/#method_throw).